Zotero/Scenes/Detail/ItemDetail/Views/ItemDetailViewController.swift

```diff
@@ -53,36 +69,77 @@ final class ItemDetailViewController: UIViewController {
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { [weak self] _ in
+                    guard let self else { return }
+                    // Need to reload data to current state before going back to foreground. iOS reloads the collection view when going to foreground with current snapshot. When
+                    // editing text fields we don't update the snapshot (so that the cell is not reloaded while typing), so the edited fields are reset to previous state.
+                    collectionViewHandler.reloadAll(to: viewModel.state, animated: false)
+                })
+                .disposed(by: disposeBag)
+
+            controllers.userControllers?.fileDownloader.observable
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { [weak self] update in
+                    guard let self else { return }
+                    viewModel.process(action: .updateDownload(update))
+
+                    guard viewModel.state.attachmentToOpen == update.key else { return }
+
+                    // 🍎 check logic of processes
+                    switch update.kind {
+                    case .ready:
+                        viewModel.process(action: .attachmentOpened(update.key))
+                        coordinatorDelegate?.showAttachment(key: update.key, parentKey: update.parentKey, libraryId: update.libraryId)
+
+                    case .failed(let error):
+                        viewModel.process(action: .attachmentOpened(update.key))
+                        coordinatorDelegate?.showAttachmentError(error)
+
+                    case .cancelled:
+                        viewModel.process(action: .attachmentOpened(update.key))
+
+                    case .progress:
+                        return
+                    }
+                })
+                .disposed(by: self.disposeBag)
+        }
     }
 
-    override func viewDidAppear(_ animated: Bool) {
-        super.viewDidAppear(animated)
-        self.didAppear = true
+    override func viewIsAppearing(_ animated: Bool) {
+        super.viewIsAppearing(animated)
+        viewModel.process(action: .loadInitialData)
     }
 
     override func viewDidLayoutSubviews() {
         super.viewDidLayoutSubviews()
-
-        if !self.didAppear {
-            // Collapsed abstract is sometimes rendered incorrectly initially. The height of cell has proper height, but only 1 line is shown instead of 2.
-            self.collectionView.reloadData()
-        }
-
-        guard let key = self.viewModel.state.preScrolledChildKey, self.collectionViewHandler.hasRows else { return }
-        self.collectionViewHandler.scrollTo(itemKey: key, animated: false)
-        self.viewModel.process(action: .clearPreScrolledItemKey)
+        // 🍎 move to is appearing? check where this `preScrolledChildKey` is set, only during start? we can probably lose clear action
```

Comment:
This is used to focus on note/attachment when this view appears. It's set in state initially. Definitely movable to `viewIsAppearing`. The `.clearPreScrolledItemKey` probably can't be removed, otherwise on next `viewIsAppearing` it would re-focus on this key, which is not what we would want.

Reply:
`viewIsAppearing` is only called once, that's why I considered this simplification, but since the table may not have rows yet, I left it as is in the end.

Reply:
Fixed in https://github.com/zotero/zotero-ios/pull/1021/commits/4379c399244ba73fb0770afda77bc7d95f14e28b

Reply:
Wait so this didn't work in `viewIsAppearing`? You could definitely move it there and get rid of `viewDidLayoutSubviews`. I just said you probably shouldn't remove the `. clearPreScrolledItemKey` action, otherwise this would be repeated on each appearance.

Reply:
`viewIsAppearing` is called before the collection view has fully reloaded.